servatrice/src/serversocketinterface.cpp

```diff
@@ -974,6 +975,48 @@ Response::ResponseCode ServerSocketInterface::cmdBanFromServer(const Command_Ban
     return Response::RespOk;
 }
 
+Response::ResponseCode ServerSocketInterface::cmdForgotPassword(const Command_ForgotPassword &cmd)
+{
+
+	bool forgotPasswordEnabled = settingsCache->value("users/enableForgotPassword", true).toBool();
+	if (!forgotPasswordEnabled)
+		return Response::RespInternalError;
+
+	if (tooManyForgotPasswordAttempts(this->getAddress()))
+		return Response::RespInternalError;
+
+	bool requireForgotPasswordEmail = settingsCache->value("users/forgotpasswordemailrequired", true).toBool();;
+	bool requireForgotPasswordClientID = settingsCache->value("users/forgotpasswordclientidrequired", true).toBool();
+
+	QString userName = QString::fromStdString(cmd.user_name());
+	QString password = QString::fromStdString(cmd.password());
+	QString userEmail = QString::fromStdString(cmd.email());
+	QString clientID = QString::fromStdString(cmd.clientid());
+
+	//if requesting user is a trusted user, change the password for the requested user
```

Comment:
This does trigger the actual changeUserPassword DB functionality.  What other process are you referring to?  Take for example if the admin/mod wants to reset a different user account other than there own.  They would have to pass the other users information into a function that calls the DB functionality that changes the password.  That's basically what this does.  I have no problem removing the check for if there is a moderator/admin making the call and it ignoring the various other checks used to validate the user account is in fact the user making the call and work the functionality in some were else.  But logically it would be doing very similar work.